### PR TITLE
🌱 Bump golang to v1.20.10 and related change

### PR DESCRIPTION
**Pull request by mquhuy**

### CODE REVIEW

File: docs/architecture.md [hunk 1]
```diff
@@ -165,7 +165,7 @@ metadata:
     cluster.x-k8s.io/control-plane: "true"
     cluster.x-k8s.io/cluster-name: "test1"
 spec:
-  version: v1.18.0
+  version: v1.20.10
```
Comment (lentzi90) [inline docs/architecture.md:168]: This is the kubernetes version :sweat_smile:

File: docs/architecture.md [hunk 2]
```diff
@@ -209,7 +209,7 @@ spec:
     kind: Metal3Machine
     name: test1-controlplane-0
   providerID: metal3://8e16d3b6-d48c-41e0-af0f-e43dbf5ec0cd
-  version: v1.18.0
+  version: v1.20.10
```
Comment (lentzi90) [inline docs/architecture.md:212]: Same here

**Review by lentzi90 — COMMENTED** [review]
nit: remove the change to the kubernetes version in the docs. Otherwise this looks good to me :slightly_smiling_face:

File: docs/architecture.md (hunk 1 above)
Comment (mquhuy) [reply docs/architecture.md:168]: Oops. Sorry :rofl:

**Review by lentzi90 — COMMENTED** [review]
/approve

**Comment by mquhuy** [comment]
@Rozzii @smoshiur1237 @adilGhaffarDev @mboukhalfa

**Review by smoshiur1237 — APPROVED** [review]
/lgtm

**Pull request merged**